src/include/catalog/column_catalog.h

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// column_catalog.h
+//
+// Identification: src/include/catalog/column_catalog.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// pg_attribute
+//
+// Schema: (column: column_name)
+// 0: table_oid (pkey), 1: column_name (pkey), 2: column_offset, 3: column_type,
+// 4: is_inlined, 5: is_primary, 6: is_not_null
+//
+// Indexes: (index offset: indexed columns)
+// 0: table_oid & column_name (unique)
+// 1: table_oid & column_offset (unique)
+// 2: table_oid (non-unique)
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "catalog/abstract_catalog.h"
+
+namespace peloton {
+namespace catalog {
+
+class ColumnCatalog : public AbstractCatalog {
+ public:
+  // Global Singleton
+  static ColumnCatalog *GetInstance(storage::Database *pg_catalog = nullptr,
+                                    type::AbstractPool *pool = nullptr);
+
+  ~ColumnCatalog();
+
+  inline oid_t GetNextOid() { return oid_++ | COLUMN_OID_MASK; }
+
+  // Write related API
+  bool InsertColumn(oid_t table_oid, const std::string &column_name,
```

Comment:
Again, we should include a description of what the parameters mean in the function header.